src/target/cortexm.c

```diff
@@ -181,11 +181,43 @@ ADIv5_AP_t *cortexm_ap(target *t)
 
 static void cortexm_mem_read(target *t, void *dest, target_addr src, size_t len)
 {
+	/* flush data cache for RAM regions that intersect requested region */
```

Comment:
I'd pull this whole blob out into a `cortexm_cache_clean` function that can be called from read and write to avoid code duplication.